python/ray/data/tests/test_auto_parallelism.py

```diff
@@ -1,10 +1,106 @@
 import pytest
+from dataclasses import dataclass
 
 import ray
 from ray.data.context import DatasetContext
+from ray.data._internal.util import _autodetect_parallelism
 from ray.tests.conftest import *  # noqa
 
 
+@dataclass
+class TestCase:
+    avail_cpus: int
+    data_size: int
+    expected_parallelism: int
+
+
+MB = 1024 * 1024
+GB = 1024 * MB
```

Comment:
Nit: These are mebibytes and gibibytes, not megabytes and gigabytes.
```suggestion
MiB = 1024 * 1024
GiB = 1024 * MiB
```